tests/internals/update_ta/update_ta_test.go

```diff
@@ -0,0 +1,372 @@
+            - sleep
+            - "30"
+            imagePullPolicy: IfNotPresent
+        restartPolicy: Never
+    backoffLimit: 1
+  pollingInterval: 5
+  minReplicaCount: {{.MinReplicas}}
+  maxReplicaCount: {{.MaxReplicas}}
+  successfulJobsHistoryLimit: 0
+  failedJobsHistoryLimit: 0
+  triggers:
+  - type: cpu
+    metadata:
+      type: Utilization
+      value: "50"
+    authenticationRef:
+      name: {{.TriggerAuthName}}
+      kind: {{.TriggerAuthKind}}
+`
+
+	scaledJob2Template = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledJob
+metadata:
+  name: {{.ScaledJob2}}
+  namespace: {{.TestNamespace}}
+spec:
+  jobTargetRef:
+    template:
+      spec:
+        containers:
+          - name: external-executor
+            image: busybox
+            command:
+            - sleep
+            - "30"
+            imagePullPolicy: IfNotPresent
+        restartPolicy: Never
+    backoffLimit: 1
+  pollingInterval: 5
+  minReplicaCount: {{.MinReplicas}}
+  maxReplicaCount: {{.MaxReplicas}}
+  successfulJobsHistoryLimit: 0
+  failedJobsHistoryLimit: 0
+  triggers:
+  - type: cpu
+    metadata:
+      type: Utilization
+      value: "50"
+    authenticationRef:
+      name: {{.TriggerAuthName}}
+      kind: {{.TriggerAuthKind}}
+`
+)
+
+func TestTriggerAuthenticationGeneral(t *testing.T) {
+	// setup
+	t.Log("--- setting up ---")
+	// Create kubernetes resources
+	kc := GetKubernetesClient(t)
+
+	t.Log("--- testing triggerauthentication  ---")
+	data, templates := getTemplateData(triggerAuthKind)
+	CreateKubernetesResources(t, kc, namespace, data, templates)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, namespace, minReplicas, 180, 3),
+		"replica count should be %d after 3 minutes", minReplicas)
+
+	testTriggerAuthenticationStatusValue(t, data, triggerAuthKind)
+
+	// Clean resources and then testing clustertriggerauthentication
+	DeleteKubernetesResources(t, namespace, data, templates)
+
+	t.Log("--- testing clustertriggerauthentication  ---")
+	data, templates = getTemplateData(clusterTriggerAuthKind)
+	CreateKubernetesResources(t, kc, namespace, data, templates)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, namespace, minReplicas, 180, 3),
+		"replica count should be %d after 3 minutes", minReplicas)
+
+	testTriggerAuthenticationStatusValue(t, data, clusterTriggerAuthKind)
+	DeleteKubernetesResources(t, namespace, data, templates)
+}
+
+func checkScaledObjectStatusFromKubectl(t *testing.T, kind string, expected string) {
+	time.Sleep(1 * time.Second)
+	kctlGetCmd := fmt.Sprintf(`kubectl get %s/%s -n %s -o jsonpath="{.status.scaledobjects}"`, kind, triggerAuthName, namespace)
+	output, err := ExecuteCommand(kctlGetCmd)
+	assert.NoErrorf(t, err, "cannot get rollout info - %s", err)
+
+	unqoutedOutput := strings.ReplaceAll(string(output), "\"", "")
+	assert.Equal(t, expected, unqoutedOutput)
+}
+
+func checkScaleJobStatusFromKubectl(t *testing.T, kind string, expected string) {
```

Comment:
Could you please make these functions more generic (to accept name, namespace ... as parameters) and move them to https://github.com/kedacore/keda/blob/main/tests/helper/helper.go as it might be beneficial for other following tests